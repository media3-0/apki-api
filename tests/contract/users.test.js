const Lokka = require('lokka').Lokka;
const Transport = require('lokka-transport-http').Transport;

const API_URI = 'http://api:9778/graphql';

const client = new Lokka({
  transport: new Transport(API_URI),
});

describe('Queries', () => {
  describe('allUsers', () => {
    // given
    const queries = {
      'without params': `
        {
          allUsers {
              id
              nickname
              email
              roles
              posts {
                title
                content
              }
          }
        }
      `,
      'with limit': `
        {
          allUsers(limit: 2) {
              id
              nickname
              email
              roles
              posts {
                title
                content
              }
          }
        }
      `,
    };

    Object.entries(queries).map(async ([name, query]) => {
      it(name, async () => {
        // when
        const result = await client.query(query);
        // then
        expect(result).toMatchSnapshot();
      });
    });

  });
});
