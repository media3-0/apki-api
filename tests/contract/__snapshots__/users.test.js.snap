exports[`Queries allUsers with limit 1`] = `
Object {
  "allUsers": Array [
    Object {
      "email": "edziu@email",
      "id": "aaaaaaaaaaaaaaaaaaaaaaa1",
      "nickname": "Edziu",
      "posts": Array [
        Object {
          "content": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus fringilla quis ipsum ut hendrerit. Etiam rutrum elementum urna, et fringilla lacus dapibus sed.",
          "title": "Lorem ipsum",
        },
      ],
      "roles": Array [
        "student",
      ],
    },
    Object {
      "email": "janusz@email",
      "id": "aaaaaaaaaaaaaaaaaaaaaaa2",
      "nickname": "Janusz",
      "posts": Array [
        Object {
          "content": "Aliquam eu mollis felis. Maecenas quis pulvinar neque. Donec arcu ex, ultrices sit amet nunc ac, interdum dapibus nunc.",
          "title": "Maecenas mattis",
        },
      ],
      "roles": Array [
        "teacher",
      ],
    },
  ],
}
`;

exports[`Queries allUsers without params 1`] = `
Object {
  "allUsers": Array [
    Object {
      "email": "edziu@email",
      "id": "aaaaaaaaaaaaaaaaaaaaaaa1",
      "nickname": "Edziu",
      "posts": Array [
        Object {
          "content": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus fringilla quis ipsum ut hendrerit. Etiam rutrum elementum urna, et fringilla lacus dapibus sed.",
          "title": "Lorem ipsum",
        },
      ],
      "roles": Array [
        "student",
      ],
    },
    Object {
      "email": "janusz@email",
      "id": "aaaaaaaaaaaaaaaaaaaaaaa2",
      "nickname": "Janusz",
      "posts": Array [
        Object {
          "content": "Aliquam eu mollis felis. Maecenas quis pulvinar neque. Donec arcu ex, ultrices sit amet nunc ac, interdum dapibus nunc.",
          "title": "Maecenas mattis",
        },
      ],
      "roles": Array [
        "teacher",
      ],
    },
    Object {
      "email": "andrzej@email",
      "id": "aaaaaaaaaaaaaaaaaaaaaaa3",
      "nickname": "Andrzej",
      "posts": Array [
        Object {
          "content": "Suspendisse eget tempor sem, eu lobortis tortor. Donec quis sem sapien. Pellentesque nec massa elit. Curabitur sagittis nibh diam, ut volutpat tortor rutrum vel.",
          "title": "Praesent laoreet dui at dui",
        },
      ],
      "roles": Array [
        "moderator",
      ],
    },
    Object {
      "email": "zdzisiek@email",
      "id": "aaaaaaaaaaaaaaaaaaaaaaa4",
      "nickname": "Zdzisiek",
      "posts": Array [
        Object {
          "content": "In vitae congue quam. Aenean vel porttitor neque. Etiam non neque vel elit luctus euismod eget non dui. In venenatis arcu dictum elit tristique dictum. Fusce vel mauris vel lorem facilisis egestas. Sed nec interdum orci.",
          "title": "Proin suscipit non quam",
        },
        Object {
          "content": "In vel porttitor neque. Etiam non neque vel elit luctus euismod eget non dui. In venenatis arcu dictum elit tristique dictum. Fusce vel mauris vel lorem facilisis egestas. Sed nec interdum orci.",
          "title": "Sipm suscipit non quam",
        },
      ],
      "roles": Array [
        "admin",
        "teacher",
      ],
    },
  ],
}
`;
